File: src/kernelprolog/terms/SxxMachine/Sink.cpp
using namespace std;

#include "Sink.h"
#include "Prog.h"
#include "../../../machine/SxxMachine/Term.h"
#include "../../main/SxxMachine/Builtins.h"

namespace SxxMachine
{

	Sink::Sink(Prog* p) : Fluent(p)
	{
	}

	Term* Sink::collect()
	{
		return nullptr;
	}
}

File: jsrc/machine/SxxMachine/pterm/Sink.java
package SxxMachine.pterm;

import SxxMachine.Prog;
import SxxMachine.Term;
import SxxMachine.Undoable;

abstract public class Sink extends Fluent implements Undoable {

	public Sink(Prog p) {
		super(p);
	}

	abstract public int putElement(Term T);

	public Term collect() {
		return null;
	}
}
